src/seedu/addressbook/storage/StorageFile.java

```diff
@@ -83,7 +86,11 @@ private static boolean isValidPath(Path filePath) {
      *
      * @throws StorageOperationException if there were errors converting and/or storing data to file.
      */
-    public void save(AddressBook addressBook) throws StorageOperationException {
+    public void save(AddressBook addressBook, boolean shouldAlreadyExist) throws StorageOperationException {
+    	// Catch error where user deletes storage while program is running
+    	if (!path.toFile().isFile() && shouldAlreadyExist) {
+    		throw new StorageOperationException("File was deleted during program execution: " + path);
```

Comment:
As mentioned in the tutorial, this was a good attempt at handling an exception. It's good that the error message is thrown to the UI to handle, but throwing the error message this early means that the data is not saved, and the file is not recovered. It might be a better idea to use a boolean variable to store this flag, then throwing the exception at the end of this function.